lib/SPIRV/SPIRVReader.cpp

```diff
@@ -2810,11 +2810,16 @@ Function *SPIRVToLLVM::transFunction(SPIRVFunction *BF) {
   if (Loc != FuncMap.end())
     return Loc->second;
 
-  auto IsKernel = isKernel(BF);
+  bool IsKernel = isEntryPointKernel(BF);
 
   if (IsKernel) {
     // search for a previous function with the same name
     // upgrade it to a kernel and drop this if it's found
+    // TODO: Right now we're mapping the same LLVM kernel
+    // onto 2 different SPIR-V functions. Should we completely
+    // erase the entry point function from the SPIRVModule
+    // instead? E.g. to avoid duplicate translation when
+    // iterating over SPIRVModule's functions.
```

Comment:
Not sure if I got the comment right. I'd recommend not to remove the entry point, as it's being used by OpenCL runtime as the actual kernel function.

Reply:
Never mind - I've removed the comment altogether precisely because there any many cases that don't match the said condition:
> if the original kernel has already been translated within this module (as the below algorithm connects both SPIR-V functions to the same LLVM function)